backend/tournesol/tests/test_api_contributor_recommendations.py

```diff
@@ -0,0 +1,335 @@
+
+        for entity in response.data["results"]:
+            self.assertEqual(entity["is_public"], True)
+
+        response = self.client.get(
+            f"/users/{self.user2.username}/recommendations/{self.poll.name}",
+            format="json",
+        )
+
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 2)
+
+        for entity in response.data["results"]:
+            self.assertEqual(entity["is_public"], True)
+
+    def test_recommendations_are_filtered_by_pole(self):
+        new_poll = PollWithCriteriasFactory()
+
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(
+            f"/users/me/recommendations/{new_poll.name}", format="json"
+        )
+
+        self.assertEqual(response.status_code, 200)
+        # no entity must be recommended in this poll
+        self.assertEqual(response.data["count"], 0)
+
+        rating = ContributorRatingFactory(
+            poll=new_poll, user=self.user1, entity=self.entity1, is_public=True
+        )
+        ContributorRatingCriteriaScoreFactory(
+            contributor_rating=rating, criteria=new_poll.criterias_list[0], score=1
+        )
+
+        response = self.client.get(
+            f"/users/me/recommendations/{new_poll.name}", format="json"
+        )
+        # the private endpoint must return one recommendation
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 1)
+
+        response = self.client.get(
+            f"/users/{self.user1.username}/recommendations/{new_poll.name}",
+            format="json",
+        )
+        # the public endpoint must return one recommendation
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data["count"], 1)
+
+    def test_users_cant_list_with_non_existing_user(self):
+        """
+        Anonymous and authenticated users can't list recommendations of a
+        non-existing user.
+        """
+        response = self.client.get(
+            f"/users/non-existing/recommendations/{self.poll.name}", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(
+            f"/users/non-existing/recommendations/{self.poll.name}", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+    def test_users_cant_list_with_non_existing_poll(self):
+        """
+        Anonymous and authenticated users can't list recommendations of a
+        non-existing poll.
+        """
+        response = self.client.get(
+            f"/users/{self.user1.username}/recommendations/non-existing", format="json"
+        )
+        self.assertEqual(response.status_code, 404)
+
+        self.client.force_authenticate(self.user1)
+        response = self.client.get(
+            f"/users/{self.user1.username}/recommendations/non-existing", format="json"
```

Comment:
did you mean "users/me/recommendations/non-existing" ?

Reply:
I'm using the contributor public recommendations endpoint here, `/users/{self.user1.username}/` is the URL I need.

I'm making two requests in a row because I'm checking the behaviour with an anonymous user then a logged user.